src/wallet/api.rs

```diff
@@ -1267,193 +1270,237 @@ impl Wallet {
+                min_absolute_fee: MIN_FEE_RATE as u64 * tx_base_weight,
+                base_weight: tx_base_weight,
+                change_weight: change_weight.to_wu(),
+                change_cost: cost_of_change,
+                avg_input_weight,
+                avg_output_weight,
+                min_change_value: 294, // Minimal NonDust value: 294
+                excess_strategy: ExcessStrategy::ToChange,
+            };
 
-        // Calculate base weight
-        let tx_base_weight = TX_BASE_WEIGHT
-            + selected_weight
-            + MAX_SPLITS as u64 * (target_weight.to_wu() + change_weight.to_wu());
-
-        // Create coin selection options with adjusted target
-        let coin_selection_option = CoinSelectionOpt {
-            target_value: amount.to_sat().saturating_sub(selected_total),
-            target_feerate: feerate as f32,
-            long_term_feerate: Some(LONG_TERM_FEERATE),
-            min_absolute_fee: MIN_FEE_RATE as u64 * tx_base_weight,
-            base_weight: tx_base_weight,
-            change_weight: change_weight.to_wu(),
-            change_cost: cost_of_change,
-            avg_input_weight,
-            avg_output_weight,
-            min_change_value: 294, // Minimal NonDust value: 294
-            excess_strategy: ExcessStrategy::ToChange,
-        };
+            // Run coin selection on single addresses only
+            match select_coin(&single_output_groups, &coin_selection_option) {
+                Ok(selection) => {
+                    let additional_utxos: Vec<_> = selection
+                        .selected_inputs
+                        .iter()
+                        .flat_map(|&group_index| single_addresses[group_index].clone())
+                        .collect();
 
-        // Run coin selection on single addresses only
-        match select_coin(&single_output_groups, &coin_selection_option) {
-            Ok(selection) => {
-                let additional_utxos: Vec<_> = selection
-                    .selected_inputs
-                    .iter()
-                    .flat_map(|&group_index| single_addresses[group_index].clone())
-                    .collect();
-
-                // Combine pre-selected groups + coin selection results
-                let mut final_selection = selected_utxos;
-                final_selection.extend(additional_utxos);
-
-                log::info!("Coin selection concluded with {:?}", selection.waste);
-                Ok(final_selection)
+                    // Combine pre-selected groups + coin selection results
+                    let mut final_selection = selected_utxos;
+                    final_selection.extend(additional_utxos);
+
+                    log::info!("Selected {} {} UTXOs", final_selection.len(), utxo_type);
+                    return Ok(final_selection);
+                }
+                Err(_) => {
+                    // Continue to next UTXO type
+                    continue;
+                }
```

Comment:
This portion can be elaborated more with some logs. It can also include the cs error `e` in the log.

It can say something like "Coinselection of {utxo_type} utxos failed, because {e}". 

That will be more helpful to see in the log what exactly failed for what reason.